internal/packer_registry/types.bucket.go

```diff
@@ -0,0 +1,387 @@
+
+	return errs.ErrorOrNil()
+}
+
+// connect initializes a client connection to a remote HCP Packer Registry service on HCP.
+// Upon a successful connection the initialized client is persisted on the Bucket b for later usage.
+func (b *Bucket) connect() error {
+	// NOOP
+	if b == nil {
+		return nil
+	}
+
+	registryClient, err := NewClient(b.Config)
+	if err != nil {
+		return errors.New("Failed to create client connection to artifact registry: " + err.Error())
+	}
+	b.client = registryClient
+	return nil
+}
+
+func (b *Bucket) RegisterBuildForComponent(sourceName string) {
+	if b == nil {
+		return
+	}
+
+	if _, ok := b.Iteration.builds.Load(sourceName); ok {
+		return
+	}
+	b.Iteration.expectedBuilds = append(b.Iteration.expectedBuilds, sourceName)
+}
+
+func (b *Bucket) PublishBuildStatus(ctx context.Context, name string, status models.HashicorpCloudPackerBuildStatus) error {
+	// NOOP
+	if b == nil {
+		return nil
+	}
+
+	// Lets check if we have something already for this build
+	build, ok := b.Iteration.builds.Load(name)
+	if !ok {
+		return fmt.Errorf("no build for the component %q associated to the iteration %q", name, b.Iteration.ID)
+	}
+
+	buildToUpdate, ok := build.(*Build)
+	if !ok {
+		return fmt.Errorf("the build for the component %q does not appear to be a valid registry Build", name)
+	}
+
+	if buildToUpdate.ID == "" {
+		return fmt.Errorf("the build for the component %q does not have a valid id", name)
+	}
+
+	buildInput := &models.HashicorpCloudPackerUpdateBuildRequest{
+		BuildID: buildToUpdate.ID,
+		Updates: &models.HashicorpCloudPackerBuildUpdates{
+			PackerRunUUID: buildToUpdate.RunUUID,
+			Labels:        buildToUpdate.Metadata,
+			Status:        &status,
+		},
+	}
+
+	// Possible bug of being able to set DONE with no RunUUID being set.
+	if status == models.HashicorpCloudPackerBuildStatusDONE {
+		images := make([]*models.HashicorpCloudPackerImage, 0, len(buildToUpdate.PARtifacts))
+		var providerName string
+		for _, partifact := range buildToUpdate.PARtifacts {
+			if providerName == "" {
+				providerName = partifact.ProviderName
+			}
+			images = append(images, &models.HashicorpCloudPackerImage{ImageID: partifact.ID, Region: partifact.ProviderRegion})
+		}
+		buildInput.Updates.CloudProvider = providerName
+		buildInput.Updates.Images = images
+	}
+
+	_, err := UpdateBuild(ctx, b.client, buildInput)
+	if err != nil {
+		return err
+	}
+	buildToUpdate.Status = status
+	b.Iteration.builds.Store(name, buildToUpdate)
+	return nil
+}
+
+// CreateInitialBuildForIteration will create a build record on the Packer registry for named component.
+// This initial creation is needed so that Packer can properly track when an iteration is complete.
+func (b *Bucket) CreateInitialBuildForIteration(ctx context.Context, name string) error {
+
+	status := models.HashicorpCloudPackerBuildStatusUNSET
+	buildInput := &models.HashicorpCloudPackerCreateBuildRequest{
+		BucketSlug:  b.Slug,
+		Fingerprint: b.Iteration.Fingerprint,
```

Comment:
Seeing this client I think Fingerprint is kinda unnecessary. Not for the client, because the service requires it right now, but it doesn't do anything with it. It loads the iteration from `build.IterationID` provided bellow.

Reply:
Okay so once the upstream API changes and we no longer need to send I can remove.